laser_slam_ros/include/laser_slam_ros/laser_slam_worker.hpp

```diff
@@ -43,6 +43,10 @@ class LaserSlamWorker {
   // Get a filtered map and apply map separation if desired.
   void getFilteredMap(laser_slam_ros::PointCloud* filtered_map);
 
+  // Get a vector containing the point clouds that have been recorded since
+  // the last call to this method. This call clears the point cloud queue.
+  std::vector<laser_slam_ros::PointCloud> getQueuedPoints();
```

Comment:
I've would actually vote for leaving it like it is. The implementation looks just like it should be with modern c++ (well done!).  It is as efficient as it can be and clearly one step closer to the ideal : a function returns by its return value. That is what it is intended for. Almost all other solutions are optimizations. Since c++11 the idea became efficient in most including this case. So why stick to old optimizations that make things more complicated after all?
On top it is good to do it the "modern way" for the personal learning.